docs/design/adrs/2024-04-08-push-vs-pull-approach-for-search.md

```diff
@@ -0,0 +1,43 @@
+Using Push API vs Pull (Integrated Vectorization) technique to index data to Azure Search
+
+* **Status** - proposed
+* **Proposer:** @komalgrover
+* **Date:** 2024-04-08
+* **Technical Story:** [https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/321)
+
+## Context and Problem Statement
+
+There are two ways to get the data uploaded to the search index i.e using Push based approach or a Pull based approach. This repo is currently using Push based approach in which the developer has more control on how the index is created and what values are put into the index.
+We want to add Integrated vectorization (Pull based approach) and allow user to be able to use both the approaches.
+
+## Decision Drivers
+* Coexistance of the same index for both the approaches
+* Performance
+* Complexity - reduces the overhead of chunking & vectorization
+
+## Design Options
+* Option 1
```

Comment:
I think it'll be clearer if we can keep this to a short list and move the descriptions to be under the `### Options N` sections e.g:
```markdown
- Create via Bicep during deployment
- Create via admin app when first document uploaded
```